src/Service/Startup.cs

```diff
@@ -173,6 +177,15 @@ public void ConfigureServices(IServiceCollection services)
                 });
             }
 
+            if (runtimeConfigAvailable
+                && runtimeConfig?.Runtime?.Telemetry?.AzureLogAnalytics is not null
+                && runtimeConfig.Runtime.Telemetry.AzureLogAnalytics.Enabled)
```

Comment:
If you already check for Enabled or not before creating the singleton `AzureLogAnalyticsFlusherService`, why do you need to re-check if its enabled or not in its StartAsync function?

Reply:
The function `ExecuteAsync` of the FlusherService will start only if the service is created. You check if LogAnalytics is enabled in before creating the service. So, would you still need the checks in the ExecuteAsync function.